drawBot/context/imageObjectContext.py

```diff
@@ -0,0 +1,42 @@
+from PIL import Image
+import io
+import AppKit
+
+from .imageContext import ImageContext
+
+class BaseImageObjectContext(ImageContext):
+
+    def _writeDataToFile(self, data, path, options):
+        self._imageObjects = []
+        # we just need a path with a file extension
+        # nothing will be written to disk
+        super()._writeDataToFile(data, "temp.tiff", options)
```

Comment:
Are you sure it doesn't now write to a file named "temp.tiff"?

Reply:
separating writing outside the big loop helps to reuse everything from the ImageContext class

Reply:
But now there are to methods with "ToFile" in their names which may not actually write "to file". Perhaps something for a later refactoring. I appreciate the code reuse, but it's not very clear what's going on.

Reply:
true, what about removing `toFile`:

_saveImageDataToFile --> _storeImageData, _saveImageData

Reply:
method renamed to _storeImageData